### PR TITLE
Add alert for failing helmreleases deploying aws components

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/32121

We want to get alerted whenever one of the `HelmReleases` fore core components like the aws cloud-controller or the aws-ebs-csi-driver are in failed state. Currently, on `alba` there are some in this state, and we'd get paged by these

image

### Checklist

- [ ] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Comment by fiunchinho** [comment]
@giantswarm/team-atlas could you give me a bit of guidance regarding CI? It's failing.

File: helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+spec:
+  groups:
+  - name: aws-cloud-controller-manager
+    rules:
+    - alert: FluxHelmReleaseFailed
+      annotations:
+        description: |-
+        {{`Flux HelmRelease {{ $labels.name }} in ns {{ $labels.exported_namespace }} on {{ $labels.installation }}/{{ $labels.cluster_id }} is stuck in Failed state.`}}
+        opsrecipe: fluxcd-failing-helmrelease/
+      expr: gotk_reconcile_condition{type="Ready", status="False", kind="HelmRelease", cluster_type="management_cluster", exported_namespace!="flux-giantswarm", name=~".*(aws-ebs-csi-driver|cloud-provider-aws).*"} > 0
```
Comment (AndiDog) [inline helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml:20]: Maybe extend these to all critical ones i.e. add cilium, coredns, network-policies suffixes as well? And do it for every CAPI-based provider, not just CAPA, so we have an easy list that we can extend for the other providers?

**Comment by hervenicol** [comment]
> @giantswarm/team-atlas could you give me a bit of guidance regarding CI? It's failing.

Oh, I guess that's an error that's happening earlier in the tests and that's not properly caught:
```
Templating chart for provider: capi/capa
Error: YAML parse error on prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml: error converting YAML to JSON: yaml: line 24: could not find expected ':'
```

I think it's due to the indentation for the yaml multiline blocks `description` and `namespace`. You need an extra indentation for the content block.

Also, the tests will probably fail later (hopefully with a more meaningful error) because you did not write any unit tests.

File: helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+spec:
+  groups:
+  - name: aws-cloud-components
+    rules:
+    - alert: FluxHelmReleaseFailed
+      annotations:
+        description: |-
+          {{`Flux HelmRelease {{ $labels.name }} in ns {{ $labels.exported_namespace }} on {{ $labels.installation }}/{{ $labels.cluster_id }} is stuck in Failed state.`}}
+        opsrecipe: fluxcd-failing-helmrelease/
+      expr: gotk_reconcile_condition{type="Ready", status="False", kind="HelmRelease", cluster_type="management_cluster", exported_namespace!="flux-giantswarm", name=~".*(aws-ebs-csi-driver|cloud-provider-aws)"} > 0
```
Comment (njuettner) [inline helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml:20]: Just one question here it looks like `gotk_reconcile_condition` is deprecated and Flux team is suggesting to use `gotk_resource_info`, do we wanna switch here?

Comment (fiunchinho) [reply helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml:20]: I don't think we have that metric

image

Comment (njuettner) [reply helm/prometheus-rules/templates/kaas/phoenix/alerting-rules/aws-cloud-components.rules.yml:20]: Ok thanks 👍🏻, so maybe we don't run the min. required version for flux 🤔

**Review by njuettner — APPROVED** [review]

**Review by QuentinBisson — APPROVED** [review]
LGTM. If you want to add rocket ones, you could do that without issues :)

**Pull request merged**